src/test/emulators/auth/setup.ts

```diff
@@ -0,0 +1,57 @@
+import { Suite } from "mocha";
+import { useFakeTimers } from "sinon";
+import supertest = require("supertest");
+import { createApp } from "../../../emulator/auth/server";
+import { ProjectState } from "../../../emulator/auth/state";
+
+export const PROJECT_ID = "example";
+
+/**
+ * Describe a test suite about the Auth Emulator, with server setup properly.
+ * @param title the title of the test suite
+ * @param fn the callback where the suite is defined
+ * @return the mocha test suite
+ */
+export function describeAuthEmulator(
+  title: string,
+  fn: (this: Suite, utils: AuthTestUtils) => void
+): Suite {
+  // Avoid arrow functions in order to pass Mocha `this` properly in:
+  return describe(`Auth Emulator: ${title}`, function() {
+    let authApp: Express.Application;
+    beforeEach("setup or reuse auth server", async function() {
+      // eslint-disable-next-line no-invalid-this
```

Comment:
Try to explicitly type `this` if you have a type (Probably from something in mocha)
https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

Reply:
`this` is already typed in the `beforeEach` callback. The warning won't go away either even if I repeat it in my code like `async function(this: Mocha.Context)`. I think the `eslint-disable` is really needed.

Reply:
Ahh turns out we need [`@typescript-eslint/no-invalid-this`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-this.md,) instead of the plain eslint rule to achieve what you've described. I can follow up with another PR.

Reply:
https://github.com/firebase/firebase-tools/pull/2882